scripts/lsp_types.lua

```diff
@@ -18,19 +19,90 @@ local function tofile(fname, text)
   end
 end
 
-function M.gen(opt)
-  if vim.uv.fs_stat('./lsp.json') then
-    vim.fn.delete('./lsp.json')
+local function read_json(opt)
+  local uri = 'https://raw.githubusercontent.com/microsoft/language-server-protocol/gh-pages/_specifications/lsp/'
+    .. opt.version
+    .. '/metaModel/metaModel.json'
+
+  local res = vim
+    .system({
+      'curl',
+      uri,
+      '-o',
+      '-',
+    })
+    :wait()
+
+  if res.code ~= 0 then
+    print(res.stderr)
+    return
   end
-  vim.fn.system({
-    'curl',
-    'https://raw.githubusercontent.com/microsoft/lsprotocol/' .. opt.ref .. '/generator/lsp.json',
-    '-o',
-    './lsp.json',
+  return vim.json.decode(res.stdout)
+end
+
+local function gen_methods(protocol)
+  local output =
+    { '--- LSP methods. Generated by lsp_types.lua, keep at end of file.', 'protocol.Methods = {' }
+  local indent = (' '):rep(2)
+
+  vim.iter(protocol.requests):map(function(item)
```

Comment:
This is an unnecessary and confusing use of iter/map. This is not "mapping" anything, it's just iterating `protocol.requests`. Please avoid that.

Also, using `vim.spairs()` would allow the output be sorted, so that's another
reason not to use map().

Reply:
```suggestion
  for k, item in vim.spairs(protocol.requests)
```